### ISSUE TITLE
test_wal_acceptor_async.py::test_race_conditions is flaky

### ISSUE THREAD

**Issue by aome510** [neondatabase/neon#2246]
Example failed runs:

https://github.com/neondatabase/neon/runs/7739496574?check_suite_focus=true
https://github.com/neondatabase/neon/runs/7758969477?check_suite_focus=true

**Comment by petuhovskiy** [comment]
Looked at the pageserver logs in the first run, seems similar to walreceiver issues which can be fixed with #2253.

In the logs pageserver spends a minute without an active walreceiver connection:
```
2022-08-09T05:21:45.095713Z  INFO walreceiver_connection{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}: IdentifySystem { systemid: 7129742103689875055, timeline: 1, xlogpos: 0/16D5DB0, dbname: None }
2022-08-09T05:21:45.095847Z  INFO walreceiver_connection{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}: last_record_lsn 0/16D1CD8 starting replication from 0/16D1CD8, server is at 0/16D5DB0...
2022-08-09T05:21:45.096708Z  INFO walreceiver_connection{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}: caught up at LSN 0/16D5DB0
2022-08-09T05:21:45.229354Z  INFO walreceiver_connection{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}:safekeeper_handle_db: Connection closed regularly: connection closed
2022-08-09T05:21:45.229486Z  WARN wal_connection_manager{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}: WAL receiving task failed: walreceiver connection handling failure: connection closed
2022-08-09T05:22:46.033198Z  INFO wal_connection_manager{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}: Switching to new connection candidate: NewWalConnectionCandidate { safekeeper_id: NodeId(3), wal_source_connstr: "host=127.0.0.1 port=18165 options='-c ztimelineid=a7f8176ed947d05cd806ea3c12aa734d ztenantid=6141cd2f21fa4662a3720426c4cc99de'", reason: NoExistingConnection }
2022-08-09T05:22:46.033256Z  INFO walreceiver_connection{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}: connecting to host=127.0.0.1 port=18165 options='-c ztimelineid=a7f8176ed947d05cd806ea3c12aa734d ztenantid=6141cd2f21fa4662a3720426c4cc99de'
2022-08-09T05:22:46.033764Z  INFO walreceiver_connection{id=6141cd2f21fa4662a3720426c4cc99de/a7f8176ed947d05cd806ea3c12aa734d}:safekeeper_handle_db: connected!
```

**Comment by arssher** [comment]
Shouldn't repeat after latest walreceiver patches.
